pkg/beacon/relay/dkg/dkg.go

```diff
@@ -71,3 +118,57 @@ func ExecuteDKG(
 		groupPrivateKeyShare: gjkrResult.GroupPrivateKeyShare,
 	}, nil
 }
+
+func waitForDkgResultEvent(
+	dkgResultChannel chan *event.DKGResultSubmission,
+	startPublicationBlockHeight uint64,
```

Comment:
Yeah but you must consider the `SigningStateBlocks` while waiting. If you don't consider those `6` blocks you will timeout `6` block too early if DKG is published by members from the end of the queue. I can rename `SigningStateBlocks` to `PublicationBlocks` and treat this as an aggregated duration of all publication states. This way if a new step is added we won't be forced to modify `dkg.go`.

Reply:
On a happy path, we shouldn't exit earlier because `Publish` does not return before it's ready to publish or before it knows there is no chance to publish the result. On the other hand, if we add another state before the publication, my idea will no longer work. 

So let's maybe rename `SigningStateBlocks` to `PrePublicationBlocks`? `PublicationBlocks` suggests it's about how long it takes to publish. And let's document that `PrePublicationBlocks` is the total number of blocks it takes to execute all the required work to get ready for the result publication or to decide to skip the publication because there are not enough supporters for that particular result.